Sources/TarkaUI/Components/Base Components/Chip/Modifiers/TUIChip+Modifiers.swift

```diff
@@ -27,47 +27,78 @@ public extension TUIChip {
   ///   - badgeCount: This is used to display the badge view in top trailing only applicable for withButton type
   ///
   /// - Returns: A closure that returns the TUIChip
-  func style(filter: Filter, isSelected: Bool = false, badgeCount: Int? = nil) -> Self {
+  func style(filter: Filter, isSelected: Bool = false, badgeCount: Int? = nil, badgeColor: Color = .error) -> Self {
     var newView = self
-    newView.style = Style.filter(filter)
-    newView.isSelected = isSelected
-    newView.badgeCount = badgeCount
+    newView.style.chipStyle = ChipStyle.filter(filter)
+    newView.style.isSelected = isSelected
+    newView.style.badgeCount = badgeCount
+    newView.style.badgeColor = badgeColor
     return newView
   }
   
   func action(_ action: @escaping () -> Void) -> Self {
     var newView = self
-    newView.action = action
+    newView.style.action = action
```

Comment:
how can the style holds action?

Reply:
@gopinathcholan we are using many values, so I have created struct

Reply:
@gopinathcholan Updated with `inputItem`